src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -212,6 +215,19 @@ protected ServiceClient()
         /// <summary>
         /// Initializes a new instance of the ServiceClient class.
         /// </summary>
+        /// <param name="httpClient">HttpClient</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Microsoft.Reliability",
+            "CA2000:Dispose objects before losing scope",
+            Justification = "The created objects should be disposed on caller's side")]
+        protected ServiceClient(HttpClient httpClient)
+        {
+            InitializeHttpClient(httpClient, CreateRootHandler());
```

Comment:
The scenario here is that som eone has an HttpClient that they want to share with this client, this means that the HttpClient may be in a state such that it cannot be altered in many ways, including adding handelrs and setting headers. This gives us two options for this constructor overload:

- We can allow the user to pass in a rootHandler and a set of DelegatingHandlers - this would simply be so that these values would not be null, and not to allow editing
- We can simply ensure that these values are not set